draft-sharabayko-mops-srt.md

```diff
@@ -2086,15 +2086,583 @@ and RTTVar are updated by the socket based on algorithms for the sender
 When an SRT socket receives data, it updates its local RTT and RTTVar,
 which can be used for its own sender as well.
 
-## Congestion Control
+# SRT Packet Pacing and Congestion Control
+
+SRT provides certain mechanisms for the sender to get a feedback
+from the receiving side through acknowledgement (ACK) packets
+({{ctrl-pkt-ack}}). Every 10 milliseconds the sender receives
+the latest values of RTT, RTT variance, available buffer size,
+receiving rate and estimated link capacity. Upon reception of
+an NAK packet ({{ctrl-pkt-nak}}) the sender detects packet loss
+during the transmission. These mechanisms provide a solid background
+for various congestion control algorithms integration in the
+SRT protocol.
+
+Given that SRT can operate in live and file transfer modes, there are
+two groups of congestion control algorithms defined in SRT: live
+congestion control (LiveCC) and file transfer congestion control
+(FileCC).
+
+## SRT Packet Pacing and Live Congestion Control (LiveCC) {#liveCC}
+
+To ensure smooth video playback of a receiving peer, SRT in live mode
+({{transmission-mode-live}}) must control the sender's buffer level to
+prevent overfill and depletion. Pacing control module is
+designed to send packets as fast as they are submitted by the
+application while maintaining relatively stable buffer level.
+While this looks like a simple problem, the sender's constituent
+of the Automatic Repeat Request (ARQ) system between
+input and output of the SRT sender adds some complexity.
+
+SRT needs a certain amount of bandwidth overhead in order to have
+space to insert retransmitted packets without affecting the SRT
+sender's main stream output rate too much so that packets can be
+transmitted normally.
+
+It is done by adjusting the maximum allowed bandwidth MAX_BW
+({{maxbw}}) which limits the bandwidth usage by SRT and is used
+by live congestion control (LiveCC) module to calculate the minimum
+inter-sending time of consecutive packets PKT_SND_PERIOD.
+In principle, the space between packets determines where
+retransmissions can be inserted, and the overhead represents the
+available margin. There is an empiric calculation that defines the
+interval, in microseconds, between two packets to give a certain
+bitrate. It's a function of the average packet payload (which
+includes video, audio, etc.) and configured maximum bandwidth MAX_BW.
+See {{default-liveCC}} for details.
+
+In live mode, it is allowed to drop those packets that can not be delivered
+in time ({{too-late-packet-drop}}).
```

Comment:
```suggestion
In live mode, the sender is allowed to drop packets that cannot be delivered
in time ({{too-late-packet-drop}}).
```

Reply:
In case of live streaming, the sender is allowed to drop packets that cannot be delivered
in time ({{too-late-packet-drop}}).

Correct?

Reply:
Yes, with one tiny comment. The expressions "in case of" and "in the case of" in English have a subtle difference when used. Not easy to explain, but "in **the** case of" usually sounds better. Not a big deal either way.